### PR TITLE
[Do not merge until library changes are released]Augment ExecutionContext with TraceContext

### PR DESCRIPTION
**Pull request by yojagad**
Part of fix for https://github.com/Azure/azure-functions-host/issues/3747

### CODE REVIEW

File: src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
 
-import org.apache.commons.lang3.tuple.*;
-import org.apache.commons.lang3.exception.*;
+import com.microsoft.azure.functions.rpc.messages.BindingInfo;
+import com.microsoft.azure.functions.rpc.messages.InvocationRequest;
+import com.microsoft.azure.functions.rpc.messages.ParameterBinding;
+import com.microsoft.azure.functions.rpc.messages.TypedData;
```
Comment (pragnagopa) [inline src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java]: any reason for explicit imports as opposed to com.microsoft.azure.functions.rpc.messages.*

**Comment by yojagad** [comment]
> Looks good. Please ping on the PR after you
> 
> * Pull latest proto file
> * Update pom.xml to library changes that includes TraceContext
> * Test these changes with current production version of host and the host that has trace context changes

@pragnagopa Updated with the latest snapshot, proto changes. Tested the changes out with

=> The library updates alone but production worker.
=> The library updates and this version of the worker.
=> Host not sending traceContext but the library and the worker still having TraceContext changes.

**Comment by pragnagopa** [comment]
LGTM! @amamounelsayed please approve and if all is ok

**Comment by amamounelsayed** [comment]
@pragnagopa I have issue with the library. Trying to figure it out.

**Pull request merged**